internal/grpclog/prefixLogger.go

```diff
@@ -79,6 +79,34 @@ func (pl *PrefixLogger) Debugf(format string, args ...any) {
 
 }
 
+// LazyDebugf does info logging at verbose level 2, only
+// evaluating arguments if logging is enabled.
+func (pl *PrefixLogger) LazyDebugf(format string, args ...func() any) {
```

Comment:
Is this worth it?

This makes the call site:

```go
logger.LazyDebugf("blah %s %s %v", func() { blah1() }, func() { blah2() }, func() { blah3() })
```

vs.

```go
if logger.V(2) {
	logger.Infof("blah %s %s %v", blah1(), blah2(), blah3())
}
```

Yes it's "3 lines instead of 1 line" but:
1. it's not obvious that Debugf is checking V(2), and
2. the usage with Lazy is really hard to read and very long because of all the extraneous closures.

I think I'd rather just delete `Debugf` instead.

Reply:
How many occurrences of Debugf do we have?

Reply:
17 i believe